### PR TITLE
pageserver: remove attach/detach apis

### PR DESCRIPTION
**Pull request by jcsp**
## Problem

These APIs have been deprecated for some time, but were still used from test code.

Closes: https://github.com/neondatabase/neon/issues/4282

## Summary of changes

- It is still convenient to do a "tenant_attach" from a test without having to write out a location_conf body, so those test methods have been retained with implementations that call through to their location_conf equivalent.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: storage_controller/src/service.rs [hunk 1]
```diff
@@ -1231,6 +1231,13 @@ impl Service {
         &self,
         attach_req: AttachHookRequest,
     ) -> anyhow::Result<AttachHookResponse> {
+        let _tenant_lock = trace_exclusive_lock(
+            &self.tenant_op_locks,
+            attach_req.tenant_shard_id.tenant_id,
+            TenantOperations::ShardSplit,
+        )
+        .await;
```
Comment (problame) [inline storage_controller/src/service.rs:1239]: Not mentioned in the PR description & seems orthogonal.

File: test_runner/fixtures/pageserver/http.py [hunk 2]
```diff
@@ -253,39 +252,30 @@ def tenant_attach(
...
 
-        res = self.post(
-            f"http://localhost:{self.port}/v1/tenant/{tenant_id}/attach",
-            data=json.dumps(body),
-            headers={"Content-Type": "application/json"},
+        return self.tenant_location_conf(
+            tenant_id,
+            location_conf={
+                "mode": "AttachedSingle",
+                "secondary_conf": None,
```
Comment (problame) [inline test_runner/fixtures/pageserver/http.py:263]: It's confusing to me that this field is set.

**Review by problame — APPROVED** [review]
some nits / questions, no blockers

🔥

File: test_runner/fixtures/pageserver/http.py (hunk 2 above)
Comment (jcsp) [reply test_runner/fixtures/pageserver/http.py:263]: It's a default-able field on the rust side, so None is synonymous with omitting it.

File: storage_controller/src/service.rs (hunk 1 above)
Comment (jcsp) [reply storage_controller/src/service.rs:1239]: This was an opportunistic fix for a rare test failure, which should become more common  in https://github.com/neondatabase/neon/pull/8135 (but the fix is still correct before that PR)
